### ISSUE TITLE
Numpy version incompatibility

### ISSUE THREAD

**Issue by aaronhendry** [spacepy/spacepy#717]
There is an issue with installing spacepy on Python versions >3.10 due to an incompatibility with numpy version 1.21, which is listed as a requirement in `pyproject.toml`. This was tested against the main branch of the spacepy repo, since the pip version currently won't install on python 3.12 due to the deprecation and removal of the distutils package.

I was able to fix this fairly simply by changing the `[build-system]` section of the .toml file as follows:

```python
[build-system]
requires = ["setuptools", "wheel",
    "numpy ~= 1.12.0 ; python_version == '3.6'",
    "numpy ~= 1.15.0 ; python_version == '3.7' and (platform_system != 'Darwin' or platform_machine != 'arm64')",
    "numpy ~= 1.20.0 ; python_version == '3.7' and platform_system == 'Darwin' and platform_machine == 'arm64'",
    "numpy ~= 1.17.0 ; python_version == '3.8' and (platform_system != 'Darwin' or platform_machine != 'arm64')",
    "numpy ~= 1.20.0 ; python_version == '3.8' and platform_system == 'Darwin' and platform_machine == 'arm64'",
    "numpy ~= 1.18.0 ; python_version == '3.9' and (platform_system != 'Darwin' or platform_machine != 'arm64')",
    "numpy ~= 1.21.0 ; python_version == '3.9' and platform_system == 'Darwin' and platform_machine == 'arm64'",
    "numpy ~= 1.21.0 ; python_version == '3.10'",
    "numpy ~= 1.26.0 ; python_version >= '3.11'",
    ]
build-backend = "setuptools.build_meta"
```

You can also see this on my forked version of the file [here](https://github.com/aaronhendry/spacepy/blob/main/pyproject.toml). I haven't really tested this on other systems, so I didn't want to risk a pull-request. I tested it on my work system, which is running MacOS Ventura 13.5 on an M2 Pro processor; I haven't tested it on any other systems. I used numpy version 1.26 because that's the current pip version, and the version I had installed -- it may work with earlier versions of numpy as well, I haven't tested that.

**Comment by jtniehof** [comment]
Thanks Aaron. The workaround for 3.11 is to first install numpy, then use `pip install --no-build-isolation` so that the build is against the installed numpy.

I was hoping to automate more of the release process before the next release, but you're right that 3.12 forces the issue. We'll be preparing that over the next week or so.

**Comment by alexandercushen** [comment]
I believe I am having a similar issue, and the `pip install --no-build-isolation` fix does not work; I get the error message ending:

`stutils/fcompiler/gnu.py", line 272, in version_match
          v = self.gnu_version_match(version_string)
        File "/private/var/folders/l9/dnt9qhfd1psfsmj9t5zfr0j00000gq/T/pip-build-env-r8ria6mo/overlay/lib/python3.10/site-packages/numpy/distutils/fcompiler/gnu.py", line 74, in gnu_version_match
          raise ValueError(err + version_string)
ValueError: A valid Fortran version was not found in this string:
      arm64-apple-darwin20.0.0-gfortran: warning: could not understand version ‘13.00.00’
      11.3.0
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for spacepy
Failed to build spacepy
ERROR: Could not build wheels for spacepy, which is required to install pyproject.toml-based projects`

I have tried many times installing via conda and macports, and tend to get similar messages relating to gfortran or numpy. I am on macOS Ventura 13.6, with M2 processor. I have also tried using an older version of python (3.10.13), but this did not help either.
